src/app/(admin)/admin/settings/rooms/page.tsx

```diff
@@ -6,14 +6,45 @@ import Link from "next/link";
 import { api } from "~/trpc/react";
 
 const Rooms = () => {
+  const { data: rooms, isLoading } = api.rooms.list.useQuery();
   const router = useRouter();
+  console.log(rooms)
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
   return (
     <div className="flex h-full flex-col p-2">
       <div className="flex-1">
         <h3 className="text-2xl font-bold">Rooms</h3>
         <AdminButton>
-        <Link href="/admin/settings/rooms/create">Add a room</Link>
-      </AdminButton>
+          <Link href="/admin/settings/rooms/create">Add a room</Link>
+        </AdminButton>
+      </div>
+      <div>
+        <table className="w-full table-auto divide-y divide-gray-200">
+          <thead className="text-left">
+            <tr>
+              <th>Name</th>
+              <th>Description</th>
+            </tr>
+          </thead>
+          <tbody>
+            {rooms?.map(({id, name, description}) => {
+              return (
+                <tr
```

Comment:
Add an onclick attribute to the <tr> element that redirects the user to the show page for the room. Something like this should do it:

```suggestion
                <tr
                                  onClick={() => router.push(`/admin/settings/rooms/${id}`)}

```

Reply:
And done